src/core/tests/mod.rs

```diff
@@ -730,6 +789,361 @@ fn test_iterate_candidates_consider_no_estimate_tx_prob() {
     );
 }
 
+#[test]
+/// This test verifies that when a transaction is skipped, other transactions
+/// from the same address with higher nonces are not included in a block.
```

Comment:
Specifically, we want it to be the case that these higher-nonce transactions aren't even _considered_.

Reply:
No action needed on this, btw.  I'm happy to take it once I merge this to #3335

Reply:
That is what the test is checking, but you're right, the comment should be more clear. Thanks for handling that!